website/docs/components/table/partials/code/component-api.md

```diff
@@ -1,38 +1,50 @@
 ## Component API
 
-The Table component itself is where most of the options will be applied. However, the child components can also be used if a custom implementation is needed.
+The Table component itself is where most of the options will be applied. However, the APIs for the child components are also documented here, in case a custom implementation is desired.
 
-- The `Hds::Table::Tr` component is a template-only component. It supports use of [`...attributes`](https://guides.emberjs.com/release/in-depth-topics/patterns-for-components/#toc_attribute-ordering) but is not eligible to receive interactions (e.g., it cannot have an `onClick` event handler attached directly to it). It can contain `Hds::Table::Th` or `Hds::Table::Td` components.
-- The `Hds::Table::Th` component is a template-only component. It supports use of [`...attributes`](https://guides.emberjs.com/release/in-depth-topics/patterns-for-components/#toc_attribute-ordering) and can contain interactive elements but is not eligible to receive interactions itself. It‘s unlikely you‘ll need to add interactive elements, as sorting is already provided.
-- The `Hds::Table::Td` component is a template-only component. It supports use of [`...attributes`](https://guides.emberjs.com/release/in-depth-topics/patterns-for-components/#toc_attribute-ordering) and can contain interactive elements (e.g., `<td><a href="user-info.html">User info</a></td>`) but is not eligible to receive interactions itself (e.g., it cannot have an `onClick` event handler attached directly to it).
+### Table
 
 <Doc::ComponentApi as |C|>
   <C.Property @name="<:head>" @type="named block">
-    This is a named block where the content for the table head (`<thead>`) is rendered.
+    This is a named block where the content for the table head (`<thead>`) is rendered. Note: most consumers are unlikely to need to use this named block directly.
   </C.Property>
   <C.Property @name="<:body>" @type="named block">
     This is a named block where the content for the table body (`<tbody>`) is rendered.
   </C.Property>
   <C.Property @name="model" @type="array">
-    If defined, sets the data source that gets yielded by the `:body` named block.
+    Indicates the data model to be used by the table.
   </C.Property>
   <C.Property @name="columns" @type="array">
-    Use an `array` hash to define your table columns. While `key` and `label` are required, other options include `isSortable`, `align` (for text-alignment), and `width`.
+  Use a `hash` within the array to define each column.
```

Comment:
Part 1: I agree and will update the definition of the columns
Part 2: Since we've already used array hashes in other components, I will not include additional (and, tbh, non-essential) information here.

Reply:
regarding the use of `hash` as a way to explain what consumers should do: in this implementation, where is the `hash`? https://github.com/hashicorp/cloud-ui/blob/main/engines/consul/addon/components/cluster-peering/peers-table/index.hbs

Not every consumer will use an inline `hash` helper in the handlebars template, so we have to write our documentation accordingly, to take in account these two different ways of providing the `@columns` argument:
- `hash` helper in the template 
- object computed in the backing class

Reply:
We've taken really clear documentation and slowly mutated it to be very confusing. 

There are no clear lines between two very distinct approaches: 

1. What someone needs to understand in order to use the table the way we designed it to be used
2. All of the other things that will be useful if they are one of the edge cases that builds their own table from our component. 

My concern is that we have increased the risk of making all of this really confusing, when all I set out to do was correct outdated documentation. 

I also don't think I know _where_ exactly it started going off the rails, so I don't know how to get us back to something simpler and redirect future work to JIRA tickets. 

WDYT?